### DISCUSSION TITLE
Call $http.send() with a string as request body

### DISCUSSION THREAD

**Discussion by helander** [pocketbase/pocketbase#3058] [Q&A] [answered]
I need to call a remote http server from from a JS hook. The remote server requires the request body to be a string (non JSON). It looks like I can not pass an arbitrary string as request data for the send() method. Is this correct, or is there some way to achieve this?

**Comment by ganigeorgiev** [comment]
It is possible to set the `data` prop as string (_even though its type declaration is defined as map_) but it will be serialized as json, aka. `data: "test"` will result in `"test"` body.

_I'll consider allowing plain string without the json marshaling with the next release._

**Reply by helander** [reply]
If I have 
    data: "test"

I get:

> panic: GoError: json: cannot unmarshal string into Go struct field sendConfig.Data of type map[string]interface {}

**Reply by ganigeorgiev** [reply]
Oh, I forgot the initial config scanning. In this case - Yes, it is not possible to set a string at the moment. I'll consider soft-deprecating the `$http.sendConfig.data` field in favor of `body` which will be raw string.

**Reply by helander** [reply]
Ok, thanks.
Meanwhile I will use $os.exec('curl',,, ...)

**Reply by ganigeorgiev** [reply, marked as answer]
Yes, that's one possible workaround.

_I've submitted a small fix in `master` (https://github.com/pocketbase/pocketbase/commit/b1093baef760deb32e642e89556eff42dc5b4639) and it will be available with the next minor v0.17.2 release sometime later this week._

**Reply by helander** [reply]
I tested with 0.17.2 and it works fine 😀
